File: server_integration_tests/java/java_server/src/main/java/dev/graphlink/javaserver/impl/BulkCreateServiceImpl.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.inputs.CreateArticleInput;
import dev.graphlink.javaserver.generated.services.BulkCreateService;
import dev.graphlink.javaserver.generated.types.Article;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class BulkCreateServiceImpl implements BulkCreateService {

    @Override
    public int bulkCreate(List<? extends List<CreateArticleInput>> matrix) {
        int count = 0;
        return count;
    }

    @Override
    public void validateBulkCreate(List<? extends List<CreateArticleInput>> matrix) {

    }
}

File: server_integration_tests/kotlin/kotlin_server_suspend/src/main/kotlin/dev/graphlink/kotlinserversuspend/impl/BulkCreateServiceImpl.kt
package dev.graphlink.kotlinserversuspend.impl

import dev.graphlink.kotlinserversuspend.generated.inputs.CreateArticleInput
import dev.graphlink.kotlinserversuspend.generated.services.BulkCreateService
import org.springframework.stereotype.Service

@Service
class BulkCreateServiceImpl : BulkCreateService {
    override suspend fun bulkCreate(matrix: List<List<CreateArticleInput>?>?): Int = 0

    override suspend fun validateBulkCreate(matrix: List<List<CreateArticleInput>?>?) {}
}
